compiler/test/dotty/tools/dotc/ParallelTesting.scala

```diff
@@ -399,18 +399,25 @@ trait ParallelTesting {
           .takeWhile(_.getMethodName != "invoke0")
           .mkString("    ", "\n    ", "")
 
-      import java.io.ByteArrayOutputStream
+      import java.io.{ ByteArrayOutputStream, PrintStream }
       import java.net.{ URL, URLClassLoader }
 
       val printStream = new ByteArrayOutputStream
+      val oldOut = System.out
+      val oldErr = System.out
       try {
         // Do classloading magic and running here:
         val ucl = new URLClassLoader(Array(dir.toURI.toURL))
         val cls = ucl.loadClass("Test")
         val meth = cls.getMethod("main", classOf[Array[String]])
 
+        val ps = new PrintStream(printStream)
+        System.setOut(ps)
```

Comment:
this is thread-unsafe

How did this work in the CI before? how did other tests pass?